views/Products/Create/Form.tsx

```diff
@@ -1,16 +1,39 @@
-import { FC } from 'react'
-import { useForm } from 'react-hook-form'
+import Image from 'next/image'
+import { ChangeEvent, FC, useState } from 'react'
+import { SubmitHandler, useForm } from 'react-hook-form'
 
 import { MutableProductFormProps, ProductEditable } from '@/services/GraphQL/products/types'
+import stylesInput from '@/styles/EditEvent.module.scss'
+
+// import styles from './SaveorDelete.module.scss'
+
+interface ProductEditableWithImg extends ProductEditable {
+  image?: FileList
+}
 
 const CreateProductForm: FC<MutableProductFormProps> = ({ onSubmit, loading }) => {
+  const [imageFile, setImageFile] = useState<File | null>(null)
+  const [imageUrl, setImageUrl] = useState<string | null>(null)
   const {
     register,
     handleSubmit,
     reset,
     formState: { errors },
-  } = useForm<ProductEditable>()
-  const submitHandler = handleSubmit(onSubmit)
+  } = useForm<ProductEditableWithImg>()
+  const submitHandler = handleSubmit(onSubmit as unknown as SubmitHandler<ProductEditableWithImg>)
```

Comment:
Why is needed the double "as"?

Could not be just:

```suggestion
  const submitHandler = handleSubmit(onSubmit as SubmitHandler<ProductEditableWithImg>)
```

?

Reply:
`as` only will change the type if `onSubmit` is asignable to `SubmitHandler<ProductEditableWithImg>`, so, I'ts needed change type to `unknown`
[@victor-heliomar check this](https://stackoverflow.com/questions/69399211/typescript-why-does-as-unknown-as-x-work)

Reply:
Thanks for clarify, will check